src/config/ConfigBuilder.ts

```diff
@@ -383,6 +402,26 @@ export class ConfigBuilder {
         }
     }
 
+    /**
+     * Statistics properties are deprecated and metrics will be used instead. This method will ensure metrics properties
+     * are set if statistics properties are set.
+     */
+    private static ensureMetricsConfiguredIfStatisticsConfigured(config: ClientConfig) {
+        if (config.hasOwnProperty('properties') &&
+            config.properties.hasOwnProperty('hazelcast.client.statistics.enabled') &&
+            (!config.hasOwnProperty('metrics') || !config.metrics.hasOwnProperty('enabled'))) {
+            throw new RangeError('hazelcast.client.statistics.enabled property is deprecated in version 5.1. ' +
```

Comment:
Is this a breaking change? If I have this property set, do I have to change my code?

Reply:
Maybe instead of throwing we should just log a warning? WDYT? @yuce  We should not break existing code. If both configs exist the new one takes precedence